account_bank_statement_import_online/models/account_journal.py

```diff
@@ -0,0 +1,99 @@
+# Copyright 2019 Brainbean Apps (https://brainbeanapps.com)
+# Copyright 2019 Dataplug (https://dataplug.io)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+import logging
+
+from odoo import models, fields, api, _
+
+_logger = logging.getLogger(__name__)
+
+
+class AccountJournal(models.Model):
+    _inherit = 'account.journal'
+
+    online_bank_statement_provider = fields.Selection(
+        selection=lambda self: self.env[
+            'online.bank.statement.provider'
+        ]._get_available_services_technical(),
+    )
+    online_bank_statement_provider_proxy = fields.Selection(
+        string='Online Bank Statement Provider (proxy)',
+        selection=lambda self: self.env[
+            'online.bank.statement.provider'
+        ]._get_available_services(),
+    )
+    online_bank_statement_provider_id = fields.Many2one(
+        string='Statement Provider',
+        comodel_name='online.bank.statement.provider',
+        ondelete='restrict',
+        copy=False,
+    )
+    online_bank_statement_provider_proxied = fields.Boolean(
+        compute='_compute_online_bank_statement_provider_proxied',
+    )
+
+    def __get_bank_statements_available_sources(self):  # pragma: no cover
+        result = super().__get_bank_statements_available_sources()
+        result.append(('online', _('Online (OCA)')))
+        return result
+
+    @api.onchange('online_bank_statement_provider_proxy')
+    def _onchange_online_bank_statement_provider_proxy(self):
+        self.online_bank_statement_provider = (
+            self.online_bank_statement_provider_proxy
+        )
+
+    @api.multi
+    def _compute_online_bank_statement_provider_proxied(self):
+        self.online_bank_statement_provider_proxied = (
+            not self.env.user.has_group('base.group_no_one')
+        )
+
+    @api.multi
+    def _update_online_bank_statement_provider_id(self):
+        OnlineBankStatementProvider = (
+            self.env['online.bank.statement.provider']
+        )
+        for journal in self.filtered('id'):
+            old_provider_id = journal.online_bank_statement_provider_id
+            journal.online_bank_statement_provider_id = False
```

Comment:
I think there will be a lot of cases where the record for the OnlineBankStatementProvider will be unlinked, and with the same values recreated againg, causing a lot of unneeded db I/O and misleading information in date_created. Better first check wether there is an existing record that satisfies the values in the journal and only unlink+ create when needed,

Reply:
This will also force the user the redo the configurations he/she did on the statement provider.

Reply:
Thanks, valid point!